Report/sections/4_testing/unit_testing.tex

```diff
@@ -0,0 +1,106 @@
+\section{Unit Testing}
+In implementing unit testing, we have followed a few approaches.
+In general, we have attempted to follow the Act-Arrange-Assert (AAA)
+methodology, as well as the naming convention 
+\texttt{MethodName\_Scenario\_ExpectedResult}.
+
+We did not use Test-Driven Development, instead resorting to testing
+when we've finished the implementation of each part. While not ideal, it
+contributed to us catching bugs and fixing errors that we wouldn't
+otherwise have caught.
+
+To illustrate our approach to testing, we will draw examples from our
+AST Builder tests.
```

Comment:
AST Builder in texttt?

Reply:
Builder should have been 'builder'. Not referring to the class as much as the idea of it.